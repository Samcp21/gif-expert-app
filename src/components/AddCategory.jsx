import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState("");
  const onInputChange = (e) => {
    const {
      target: { value },
    } = e;
    setinputValue(value);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setinputValue("");
    onNewCategory(inputValue.trim());
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gif's"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
