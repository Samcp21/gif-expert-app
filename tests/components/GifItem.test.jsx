import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem/>", () => {
  const title = "Dragon ball";
  const url =
    "https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=46f5858dmmeg5wh2wigtfmbc7mkh2ayb7uyfm72hbjxlzyqt&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  test("Match con el Snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    //Se hace una captura del codigo
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    //Acceder a la etiqueta img
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    //toBeTruthy existe el elemento
    expect(screen.getByText(title)).toBeTruthy();
  });
});
