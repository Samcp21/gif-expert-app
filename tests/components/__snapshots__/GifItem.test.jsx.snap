// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=46f5858dmmeg5wh2wigtfmbc7mkh2ayb7uyfm72hbjxlzyqt&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
